src/objects.c

```diff
@@ -1306,31 +1306,25 @@ int p11prov_obj_export_public_rsa_key(P11PROV_OBJ *obj, OSSL_CALLBACK *cb_fn,
     return ret;
 }
 
-#define EC_PUB_ATTRS 2
-int p11prov_obj_export_public_ec_key(P11PROV_OBJ *obj, OSSL_CALLBACK *cb_fn,
-                                     void *cb_arg)
+const char *p11prov_obj_get_ec_group_name(P11PROV_OBJ *obj)
 {
-    CK_ATTRIBUTE attrs[EC_PUB_ATTRS] = { 0 };
-    OSSL_PARAM params[EC_PUB_ATTRS + 1];
-    ASN1_OCTET_STRING *octet = NULL;
+    CK_ATTRIBUTE attrs[1] = { 0 };
     EC_GROUP *group = NULL;
     const unsigned char *val;
     const char *curve_name;
     int curve_nid;
     CK_RV rv;
-    int ret;
 
-    if (p11prov_obj_get_key_type(obj) != CKK_EC) {
-        return RET_OSSL_ERR;
+    if (p11prov_ctx_allow_export(obj->ctx) & DISALLOW_EXPORT_PUBLIC) {
```

Comment:
This is wrong here

Reply:
oh, yeah. Part of the rebase pains. Will fix.